docs/signing-spec.md

```diff
@@ -22,22 +30,59 @@ Asymmetric cryptography is nice for this because a signer can distribute a publi
+(In other words, all bindles MUST be signed by a `creator` key before they can be hosted.)
+When a new bindle is submitted, a Bindle server MUST reject an invoice that is signed by a creator's key for which the server does not have an entry in its keyring.
+In other words, the server MUST have the keys for every creator whose Bindle it accepts to host.
+A bindle server MUST NOT sign a bindle with its host key if it does not have the key of the creator in its keyring.
+This does not necessarily preclude a "caching proxy", where the proxy is repeating a Bindle that has been signed by another set of creator and host keys.
+
+> A _caching proxy_ is a Bindle server that takes a user's request for a bindle, finds that bindle remotely, and then caches that bindle's content locally so that it may re-serve that content at a later date. Such servers MAY sign with a proxy key.
+
+A client SHOULD reject an invoice that is not signed by a known key with the `creator` or `approver` role. A non-normative recommendation would be that a client ought to reject any bindle where neither a creator's nor a approver's signature could be verified.
+
+### The Approver Role
+
+The `approver` role describes a signer who asserts that the signer has employed a method of assessing whether the bindle is correct.
+
+Examples of approvers might include:
+
+- A developer who reviews the code and build tools for the bindle's parcels
+- A security scanner that runs an audit of the bindle and parcels
+- A security researcher who performs a security audit and deems the bindle and parcels safe.
+
+The `approver` role is specific to the security context. This signer is not deciding whether text content is editorially correct or that an image looks good. The signer is asserting that the security profile of the bindle is deemed trustworthy. That is, a vulnerability scanner is asserting, upon signature, that it detected no vulnerabilities in the bindle.
+
+An underlying assumption of the present model is that keys with the `approver` role will be vetted more carefully before inclusion in the keyring.
+End users in particular should be careful to only include well-known entities in their `approver` keyring.
 
-Bindle servers MUST add a `host` entry to a bindle upon receiving the bindle from a client.
+Verification may form a trust proxy. That is, a client may decide that if the `creator` is unknown, the bindle can still be trusted if one or more of the `approver` keys is known.
+
+### The Host role
+
+The `host` role denotes that the signer is the Bindle server that accepted the invoice from the `creator`.
+
+Bindle servers MUST add a `host` entry to a bindle upon receiving the bindle from a client and verifying the creator key.
 A bindle server MUST NOT sign the same invoice more than once with the same key.
-Clients MAY reject invoices that are signed by unknown hosts, but they also MAY allow them.
+Clients MAY reject invoices that are signed by unknown hosts.
```

Comment:
🤔 This makes me wonder if we should have an endpoint on bindle servers that allows you to fetch its private key

Reply:
Private key? Definitely not. But maybe public key. I haven't tackled key management yet because it is really tricky.

Reply:
Oh sorry, that was a total typo. Meant to say public key